### ISSUE TITLE
confusing lack of full name for StdRng on guide-rngs.html

### ISSUE THREAD

**Issue by Artoria2e5** [rust-random/book#58]
The table in https://rust-random.github.io/book/guide-rngs.html#cryptographically-secure-pseudo-random-number-generators-csprngs lists StdRng with known properties for "performance", "initialization", "memory", "security (predictability)", "forward secrecy", but notably does NOT give its full name: ChaCha12. This is confusing.

Like, what are we trying to do here? If we don't want to tie the choice of RNG down to ChaCha12, we would have no definite answers for "performance", "initialization", "memory", etc. either. If we have definite answers for these columns, it naturally follows that we definitely know what generator it is using.

While we're at it, it's a little fishy that ChaCha12 is running slower than ChaCha20. Is this a fluke in the benchmark, outdated table, or some real performance issue?

**Comment by dhardy** [comment]
ChaCha12 should be faster... those benchmarks probably are a little fishy.

We [previously](https://github.com/rust-random/rand/issues/932) used ChaCha20 (and before that HC-128, and [before that](https://github.com/rust-random/rand/pull/277) Isaac RNG), so we *have* changed the generator while roughly retaining those properties — except for improving memory usage and init speed.

Also, there are unresolved questions (#57) and the book probably does have some flaws.

**Comment by vks** [comment]
I agree that it's a bit confusing, also in comparison with the non-CSPRNG section:
* `SmallRng` mentions a incorrect state size of 16 bytes, which is probably a relic from when it was `Pcg64Mcg`. Now it's being compared to it's identical twin, `Xoshiro256PlusPlus`.
* `StdRng` stats are correct, but it's identical twin, `ChaCha12Rng` is **not** in the list.

How strongly do we care about keeping the `StdRng`/`SmallRng` algorithms "secret"? It feels a bit silly, given that we make statements about performance and state size...

I think we should do one of the following:
1. Don't make any statements about `StdRng` and `SmallRng` that we don't want to guarantee. In practice, this means removing statements about state size and performance (or only specify bounds like for the period). Instead, we could include their "identical twins" with full information.
2. Disclose the underlying algorithm. We could remove their identical twins, but I think it's sensible advertise them for use cases that require reproducibility.

**Comment by dhardy** [comment]
I concur with your assessment @vks.
